docs/ebpf/concepts/rlimit.md

```diff
@@ -0,0 +1,88 @@
+# Resource Limits
+
+Creating eBPF objects (Maps, Programs, even BTF blobs) requires kernel memory
+allocation. Before kernel version 5.11, you could control the memory available
+to a process for creating eBPF objects by adjusting the process'
+`RLIMIT_MEMLOCK` rlimit value, visible through the `ulimit -l` command.
+
+!!! note ""
+    User space typically does *not* allocate or `mlock(2)` memory for the
+    eBPF subsystem. The choice of `RLIMIT_MEMLOCK` was made early on due to its
+    connection to memory management. Unfortunately, this approach had
+    significant drawbacks, as explained in the [cover
+    letter](https://lore.kernel.org/bpf/20201201215900.3569844-1-guro@fb.com) of
+    the cgroup accounting introduced in 5.11.
+
+Starting with [version
+5.11](https://lore.kernel.org/bpf/20201201215900.3569844-1-guro@fb.com), the
+Linux kernel switched from rlimits to memory cgroup (memcg) accounting for
+managing memory limits on processes handling eBPF objects in the kernel. eBPF
+object allocations are tracked alongside regular allocations within the cgroup.
+Memory consumption and limits can be queried and set through cgroupfs, the same
+mechanism used for setting memory limits on containers.
+
+## Purpose of package `rlimit`
```

Comment:
Could also put this in the "portable eBPF" guide?

Reply:
Yep, I suggest we link to this page from the portability guide. We'll need to lean on the rlimit package to tie everything together in an example, no worries. :+1: